src/OpenTelemetry.Instrumentation.StackExchangeRedis/TracerProviderBuilderExtensions.cs

```diff
@@ -111,6 +111,30 @@ public static class TracerProviderBuilderExtensions
         string? name,
         IConnectionMultiplexer? connection,
         Action<StackExchangeRedisInstrumentationOptions>? configure)
+    {
+        return builder.AddRedisInstrumentation(name, connection, configure, null);
+    }
+
+    /// <summary>
+    /// Enables automatic data collection of outgoing requests to Redis.
+    /// </summary>
+    /// <remarks>
+    /// Note: If an <see cref="IConnectionMultiplexer"/> is not supplied
+    /// using the <paramref name="connection"/> parameter it will be
+    /// resolved using the application <see cref="IServiceProvider"/>.
+    /// </remarks>
```

Comment:
nit: We now have this duplicated with the method above. What I would suggest is use `inheritdoc` to share it.

To modify the one above to reference the new uber one which has all the parameters switch line 99 to be:

`/// <remarks><inheritdoc cref="AddRedisInstrumentation(TracerProviderBuilder, string?, IConnectionMultiplexer?, Action{StackExchangeRedisInstrumentationOptions}?, object?)" path="/remarks"/></remarks>`

Reply:
Actually on second thought, I think we could just modify the existing uber one. That would be a breaking change but the project is still prerelease so that is OK to do. No one should really be calling the uber one, it is just there for named options which is rare. I suggest these modifications...

```diff
public static class TracerProviderBuilderExtensions
{
+    public static TracerProviderBuilder AddRedisInstrumentation(
+        this TracerProviderBuilder builder,
+        object serviceKey)
+    {
+        Guard.ThrowIfNull(serviceKey);
+
+        return AddRedisInstrumentation(builder, name: null, connection: null, configure: null, serviceKey);
+    }

+    public static TracerProviderBuilder AddRedisInstrumentation(
+        this TracerProviderBuilder builder,
+        object serviceKey,
+        Action<StackExchangeRedisInstrumentationOptions> configure)
+    {
+        Guard.ThrowIfNull(serviceKey);
+        Guard.ThrowIfNull(configure);
+
+        return AddRedisInstrumentation(builder, name: null, connection: null, configure, serviceKey);
+    }

    public static TracerProviderBuilder AddRedisInstrumentation(
        this TracerProviderBuilder builder,
        string? name,
        IConnectionMultiplexer? connection,
        Action<StackExchangeRedisInstrumentationOptions>? configure,
+       object? serviceKey)
+    {
+         // Implementation not shown
+    }
}
```

The idea is to add overloads for the DI case (where `IConnectionMultiplexer` is not passed directly) which allow passing `serviceKey`.